runtime/src/main/java/com/flipkart/flux/deploymentunit/ExecutableRegistryPopulator.java

```diff
@@ -0,0 +1,146 @@
+                //for every task method found in the deployment unit create an executable and keep it in executable registry
+                for(Method method : taskMethods) {
+                    String taskIdentifier = new MethodId(method).toString();
+                    Annotation taskAnnotation = method.getAnnotationsByType(TaskClass)[0];
+                    Class<? extends Annotation> annotationType = taskAnnotation.annotationType();
+                    long timeout = 1000l; //default timeout
+                    for (Method annotationMethod : annotationType.getDeclaredMethods()) {
+                        Object value = annotationMethod.invoke(taskAnnotation, (Object[])null);
+                        if(annotationMethod.getName().equals("timeout")) { //todo: find a way get Task.timeout() name
+                            timeout = (Long) value;
+                        }
+                    }
+
+                    Object singletonMethodOwner = getClassInstanceMethod.invoke(injectorClassInstance, method.getDeclaringClass());
+                    executableRegistry.registerTask(taskIdentifier, new ExecutableImpl(singletonMethodOwner, method, timeout, classLoader));
+
+                    //add the router name to deployment unit routers
+                    routerNames.add(method.getDeclaringClass().getName() + "_" + method.getName()); //convention: task router name is classFQN_taskMethodName
+                }
+            } catch (Exception e) {
+                LOGGER.error("Unable to populate Executable Registry for deployment unit: "+deploymentUnitName+". Exception: "+e.getMessage());
+                throw new FluxError(FluxError.ErrorType.runtime, "Unable to populate Executable Registry for deployment unit: "+deploymentUnitName, e);
+            }
+        }
+
+    }
+
+    /**
+     * Loads {@Link ClassLoaderInjector} class into given deployment unit's class loader and returns it.
+     * @param deploymentUnitClassLoader
+     * @return ClassLoaderInjector class
+     */
+    private Class loadClassLoaderInjector(ClassLoader deploymentUnitClassLoader) {
+        try {
+            //Convert the class into bytes
+            byte[] classBytes = IOUtils.toByteArray(this.getClass().getResourceAsStream("/com/flipkart/flux/deploymentunit/ClassLoaderInjector.class"));
+
+            Class classLoaderInjectorClass;
+            Class classLoaderClass = Class.forName("java.lang.ClassLoader");
+            java.lang.reflect.Method method = classLoaderClass.getDeclaredMethod("defineClass", String.class, byte[].class, int.class, int.class);
+
+            // protected method invocation
+            method.setAccessible(true);
+            try {
+                Object[] args = new Object[]{ClassLoaderInjector.class.getCanonicalName(), classBytes, 0, classBytes.length};
+                classLoaderInjectorClass = (Class) method.invoke(deploymentUnitClassLoader, args);
```

Comment:
i didn't get understand. I'm saying to invoke defineClass from "deploymentUnitClassLoader" -- the param which is passed to this method. The method is ultimately invoked against that parameter.

To circumvent the permissions issue, we can extend the classLoader (or UrlClassLoader) with the relevant methods being public. Check http://download.forge.objectweb.org/asm/asm4-guide.pdf (page 17)